llama_index/embeddings/sagemaker_embedding_endpoint.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, Dict, List, Optional
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.callbacks.base import CallbackManager
+from llama_index.constants import DEFAULT_EMBED_BATCH_SIZE
+from llama_index.core.embeddings.base import BaseEmbedding, Embedding
+from llama_index.embeddings.sagemaker_embedding_endpoint_utils import (
+    BaseIOHandler,
+    IOHandler,
+)
+from llama_index.types import PydanticProgramMode
+from llama_index.utilities.aws_utils import get_aws_service_client
+
+DEFAULT_IO_HANDLER = IOHandler()
+
+
+class SageMakerEmbeddingEndPoint(BaseEmbedding):
```

Comment:
tbh, I don't love these class names. I would have preferred `SagemakerLLM` and `SagemakerEmbedding` for example. I might push a change to fix this 😆

Reply:
feel free to do so :)

Reply:
done (and kept the old LLM name for backwards compat)